components/mediation/inbound-endpoints/org.wso2.micro.integrator.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java

```diff
@@ -108,8 +111,12 @@ public static void loadInternalApis(String apiFilePath) {
                             if (name == null || name.isEmpty()) {
                                 handleException("Name not specified in one or more handlers");
                             }
+                            // Readiness-probe api is enabled by default. ( to disable use -DdisableReadiness=true )
+                            // other APIs must be enabled manually.
                             if (!Boolean.parseBoolean(
-                                    System.getProperty(Constants.PREFIX_TO_ENABLE_INTERNAL_APIS + name))) {
+                                    System.getProperty(Constants.PREFIX_TO_ENABLE_INTERNAL_APIS + name)) &&
+                                    !(READINESS_PROBE_API_NAME.equals(name) && !Boolean.parseBoolean(
```

Comment:
Instead of complicating here , shall we just add to startup script without touching this code base as discussed offline